### ISSUE TITLE
Crash v3.0.8788.4261 Git-75b931fc ( also v3.0.8789.4262 Git-04479f7d and v3.0.8790.4263 Git-8afff310) if try open Hacker menu

### ISSUE THREAD

**Issue by VictorVG** [processhacker/processhacker#944]
Crash v3.0.8788.4261 Git-75b931fc ( also v3.0.8789.4262 Git-04479f7d and v3.0.8790.4263 Git-8afff310) if try open Hacker menu

OS Win7 SP1 x64.

If i try open Hacker menu then v3.0.8789.4262 Git-04479f7d (also v3.0.8790.4263 Git-8afff310) is crashed and debugger show messages "_Unhandled exception at address 0x00000000xxxx (ntdll.dll) in ProcessHacker: .exe 0xC0000225: Object not found._" for .\ProcessHacker\settings.c string 373 as crashed location.

for v3.0.8789.4262 Git-04479f7d:

![ph](https://user-images.githubusercontent.com/906886/129282036-9e6dbc56-5380-4d2c-85ee-6dc8017d3f95.png)

for v3.0.8790.4263 Git-8afff310:

![ph2](https://user-images.githubusercontent.com/906886/129282044-2f9369cc-08b7-46ac-8927-3df2bb731b40.png)

My crash dumps needed?

P.S.

Possible crashed in to \ProcessHacker\mainwnd.c :: 1854 : 
```
    if (PhPluginsEnabled)
    {
        PH_PLUGIN_MENU_INFORMATION pluginMenuInfo;

        PhPluginInitializeMenuInfo(&pluginMenuInfo, menu, WindowHandle, PH_PLUGIN_MENU_DISALLOW_HOOKS);
        pluginMenuInfo.u.MainMenu.SubMenuIndex = Index;
        PhInvokeCallback(PhGetGeneralCallback(GeneralCallbackMainMenuInitializing), &pluginMenuInfo);
    }
```
????

**Comment by dmex** [comment]
> Crash v3.0.4261

The latest official build is v3.0.4234... your 4261 build is compiled from sources which are incomplete.

**Comment by VictorVG** [comment]
Issues is close, but crash exist:

![4254](https://user-images.githubusercontent.com/906886/129403953-59d80004-6487-4d84-95db-d0a00fd3d9c9.png)

**Comment by dmex** [comment]
> Issues is close, but crash exist

The files have other changes which haven't been completed... It'll get fixed when those other changes are published.

**Comment by VictorVG** [comment]
Ok! Understood. I'll wait, otherwise I thought I had an accident again - Friday, the 13th - "unlucky day." As Leonid Sobolev wrote in his story "Friday 13":

_- Friday, the thirteenth unlucky day. This belief came from the British - by Friday, every decent maryak gets so drunk that he cannot distinguish the bow from the stern, and the thirteenth additionally affects the psyche and here it no longer smacks of an accident but of death ..._ :)

**Comment by VictorVG** [comment]
_Git [d4be1406] 2021-08-14 06:50:12 +1000 dmex : Fix start performance monitor and restart to firmware menus on Win7_

try test open Hacker menu, but crash in _exception 0x00000225 - Required object not exist_

![4268](https://user-images.githubusercontent.com/906886/129419150-c3aa9735-7b99-4a81-8ad6-64caa886789d.png)

**Comment by VictorVG** [comment]
v3.0.8798.4271 Git-b6644725 also crashed in to new dynamic Hacker menu code:

![4271](https://user-images.githubusercontent.com/906886/129452153-51905775-13a1-4f9c-8c79-615a79db93b8.png)

For experiment, in order to exclude the possible influence of plugins, I specially minimized the set of files in the test copy to ProcessHacker.exe, ProcessHacker.exe.settings.xml (initial length 0 bytes), ProcessHacker.pdb, ProcessHacker.sig, capslist.txt, etwguids.txt , usernotesdb.xml (initial value is "empty"), but a drop of 100% is reproducible.

Therefore, after looking at the code, I see this option:

if IsOsVersion >= Win8 then call: new_dynamic_menu_code else goto: legacy_menu_code end
: legacy_menu_code
// the same code as before v3.0.8787.4260 Git-26fbe412 inclusive;

and in the new_dynamic_menu_code block, combine all the code that works with dynamic menus, for example, as a separate subroutine, possibly putting it into a separate source file for ease of development.

Now the algorithm, having checked the OS version in one place in others, does not perform this check and always adds items to the Hacker menu indicating non-existent objects, which leads to a failure.

[**PH_v3.0.8798.4271_Git-b6644725_x64_crash.zip**](https://www.upload.ee/files/13393124/PH_v3.0.8798.4271_Git-b6644725_x64_crash.zip.html)

**Comment by VictorVG** [comment]
Also **skipik** (he use OS Windows 10/11 Latest ) X6415.08.2021 speech me in PM on forum.ru-board.com:

> "The pH also drops, and not only calling the main menu, but also if you make a PCM on the tray icon. That is, the pH is minimized and we are trying to call the context menu."

**Comment by VictorVG** [comment]
Also test - v3.0.8798.4271 Git-b6644725, OS Win7 SP1 x64, Win10 21H1 (test by** skipik**), PH minimize in to try. Goto mini Window, if I press KBD Apps or use RMsClick in to this window - the have crash this exception code 0x00000225 "_Object not found_". 

Possible this crash have some source like crash if try open Hacker menu.

**Comment by VictorVG** [comment]
v3.0.8799.4272 Git-31e7947f - the mini tray window worked without errors, but the crash when trying to open the Hacker menu was saved .

**Comment by VictorVG** [comment]
**v3.0.8803.4276 Git-6a26e9ca** - issues is closed, but bug is exist - if I try open Hacker menu then crash this exception _0x00000225 Object not found_ ....

**Comment by dmex** [comment]
> issues is closed, but bug is exist

Debug assertions that were added manually are not bugs. These are features being developed and not complete. I asm still working on those menus so they'll keep asserting until they're completed.

> I try open Hacker menu then crash this exception 0x00000225 Object not found 

You're compiling source code with bleeding edge changes that I haven't had time to complete... These are not crashes, they're debug assertions required for source development.

There haven't been any official releases of the nightly builds or otherwise... If you need a stable version of the source then use revisions with a green tickbox:

![image](https://user-images.githubusercontent.com/1306177/130023577-d2daac41-146c-470d-a926-03a3a5e1efca.png)

If you don't know what assertions are or how to use them then only use revisions with those green ticks otherwise expect problems like assertions and debug statements and whatever.

**Comment by VictorVG** [comment]
Ok! I wait...

**Comment by VictorVG** [comment]
Big thanks! This issues fixed in to v3.0.8817.4290 Git e068524d.
